calyx-opt/src/passes/compile_static.rs

```diff
@@ -239,6 +239,81 @@ impl CompileStatic {
             early_reset_group.borrow().attributes.clone();
         g
     }
+
+    fn get_reset_group_name(&self, sc: &mut ir::StaticControl) -> &ir::Id {
+        // assume that there are only static enables left.
+        // if there are any other type of static control, then error out.
+        let ir::StaticControl::Enable(s) = sc else {
+            unreachable!("Non-Enable Static Control should have been compiled away. Run {} to do this", crate::passes::StaticInliner::name());
```

Comment:
`unreachable!` will signal a panic in the compiler but we should probably return an error using `Err(...)` instead since this seems like an error that can totally occur for good reason.

My heuristic here is:
1. If the error should never occur, use `unreachable`
2. If the error occurs because of a malformed user input, in this case bad compilation pipeline configuration, then show a meaningful error message